manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala

```diff
@@ -40,6 +43,32 @@ class DbClient private[db] (transactor: Transactor[IO])(
       logger.error(err)("DB status check hit error").as(false)
     }
   }
+
+  /**
+    * Record a new queue resource in the DB.
+    *
+    * Will fail if a queue already exists with the input's name.
+    */
+  def insertQueue(queue: Queue): IO[Unit] =
+    sql"""insert into queues (name, request_topic, response_topic, request_schema)
+          values (${queue.name}, ${queue.requestTopic}, ${queue.responseTopic}, ${queue.schema})""".update.run.void
+      .transact(transactor)
+
+  /**
+    * Remove the queue resource with the given name from the DB.
+    *
+    * No-ops if no such queue exists in the DB.
+    */
+  def deleteQueue(name: String): IO[Unit] =
+    sql"""delete from queues where name = $name""".update.run.void.transact(transactor)
+
+  /** Pull the queue resource with the given name from the DB, if it exists. */
+  def lookupQueue(name: String): IO[Option[Queue]] =
```

Comment:
How often are these queries likely to be run? Does it make sense to do prepared statements?

Reply:
I'll look into how gnarly doobie's low-level API is. It does compile these down to prepared statements before sending them off, so hopefully it won't be too bad to break apart.